flow/project.py

```diff
@@ -301,8 +301,9 @@ def __init__(self, id, name, jobs, cmd, directives=None):
         self.directives._keys_set_by_user = keys_set_by_user
 
     def __str__(self):
-        assert len(self._jobs) == 1
-        return f"{self.name}({str(self._jobs[0])})"
+        concat_jobs_str = str(self._jobs[0]) if len(self._jobs) == 1 else \
+            f'({str(self._jobs[0])}-{str(self._jobs[-1])})'
+        return f"{self.name}-{len(self._jobs)}: ({concat_jobs_str})"
```

Comment:
```suggestion
        assert len(self.jobs) > 0
        max_len = 3
        shown = self.jobs[:max_len-2] + ['...'] + self.jobs[-1:] if len(self.jobs) > max_len else self.jobs
        return f"{self.name}[#{len(self.jobs)}]({', '.join([str(element)[:8] for element in shown])})"
```
I do not like this representation, because it is very unusual within the Python ecosystem. I would very much prefer to follow standard conventions here and use a representation that is inspired by the representation of large numpy arrays.
```
my-op[#1](26021048)
my-op[#2](26021048, 42b7b4f2)
my-op[#3](26021048, 42b7b4f2, 44550aef)
my-op[#4](26021048, ..., 4b893796)
```
We could even consider to omit the length indication (`#X`), since a user could always use `print(len(op))` to obtain that information. I am not sure how helpful that is in practice.

---
Code to generate above output:
```python
from dataclasses import dataclass
from typing import List

import signac

@dataclass
class AgOp:
    name: str
    jobs: List[signac.contrib.job.Job]

    def __str__(self):
        assert len(self.jobs) > 0
        max_len = 3
        shown = self.jobs[:max_len-2] + ['...'] + self.jobs[-1:] if len(self.jobs) > max_len else self.jobs
        return f"{self.name}[#{len(self.jobs)}]({', '.join([str(element)[:8] for element in shown])})"

jobs = list(signac.get_project())
for i in range(min(4, len(jobs))):
    print(AgOp(name='my-op', jobs=jobs[:i+1]))
```

Reply:
@csadorf we could drop the length part but since `_JobOperation` is private, I'm not sure how a user would know the length of it's instance

Reply:
I don't follow the logic. If this class is private why would any user care about `__str__` then?

Reply:
Also, if you're talking about the jobs passed in the operation function then yes I agree that users could do that but if we provide a little information from our side then I guess it'll save an extra print statement from the users. Please suggest.

Regarding the `__str__` method, this output is printed out in the scripts

Reply:
Ok, so then let's keep the length indicator. Do you like my proposed representation otherwise?

Reply:
Also I think we should print the full job ids in this case and not limit it to 8 characters. I don't think the output will be too long for 3 jobs.
It's not necessarily unique. @bdice once suggested me to use `min_len_unique_id` method. I could use that method if you agree.

Reply:
I think using `min_len_unique_id` would be ideal.